### DISCUSSION TITLE
Rollback action

### DISCUSSION THREAD

**Discussion by 0xElectric** [craftcms/generator#7] [Ideas] [answered]
Is it possible to have rollback action to recent commands done using the generator?
For example I created a module, that action add files to `modules/module-name` and adds code the `config/app.php`, etc.
Would be handy if there is a  way to undo everything from recent action (using `craft generator` commands), instead of just deleting the files generated by `craft generator`.

(Dunno, maybe I'm dumb)

**Comment by brandonkelly** [comment, marked as answer]
If you hadn‘t made any additional uncommitted changes before running the Generator command, you can run `git reset --hard HEAD` to reset your local files to the latest Git commit.

I don’t think there’s a reliable way we could add an `undo`/`rollback` command directly to Generator.
